src/utils.ts

```diff
@@ -0,0 +1,19 @@
+import { Predicate } from 'fp-ts/lib/function';
+
+/* eslint-disable @typescript-eslint/array-type */
+export const takeUntil = <A = unknown>(predicate: Predicate<A>) => (
```

Comment:
If implemented a `find` function this can be a one-liner btw, ES6 equivalent `as.slice(0, as.findIndex(predicate) + 1)` - but since `find` isn't available in all runtimes would need to implement it

Reply:
Yes find could work as well.

I took a quick look at [how takeWhile is implemented in fp-ts](https://github.com/gcanti/fp-ts/blob/6ed09824ec98c2fd3a5dbe2b44097792ceaada9e/src/ReadonlyArray.ts#L520-L552) and decided to go with that approach.